romancal/resample/resample_utils.py

```diff
@@ -102,3 +102,31 @@ def make_output_wcs(
     )
 
     return wcs, pscale * 3600.0, pscale_ratio
+
+
+def compute_var_sky(model) -> None:
+    """
+    Add sky variance array to a datamodel.
+
+    Parameters
+    ----------
+    model : `ImageModel`
+        A datamodel to which the sky variance array will be added.
+
+    Returns
+    -------
+    None
+    """
+
+    # convert NaNs to zeros
+    data = np.nan_to_num(model["data"], nan=0.0)
+    median_data = np.median(data)
+    ok_data = model["data"] != 0
```

Comment:
For computing this median robustly the usual approach would be:
dnu = (model['dq'] & pixel.DO_NOT_USE) != 0
median_data = np.median(data[~dnu])
We should probably treat it as a bug if there is a NaN in data that isn't flagged with DO_NOT_USE.  If some test data or something triggers that, though, and we want to be robust to it, then replacing np.median with np.nanmedian would be the usual route.  The danger of converting to zero and then taking the median is that the median gets biased toward zero by the data you don't want to be considering.